gdx/src/com/badlogic/gdx/graphics/g2d/DistanceFieldBitmapFont.java

```diff
@@ -0,0 +1,192 @@
+	}
+
+	public DistanceFieldBitmapFont (FileHandle fontFile, TextureRegion region) {
+		super(fontFile, region);
+	}
+
+	public DistanceFieldBitmapFont (FileHandle fontFile) {
+		super(fontFile);
+	}
+
+	@Override
+	public BitmapFontCache newFontCache () {
+		for (int i = 0; i < regions.size; i++) {
+			getRegion(i).getTexture().setFilter(TextureFilter.Linear, TextureFilter.Linear);
+		}
+		
+		return new DistanceFieldFontCache(this, integer);
+	}
+	
+	/**
+	 * @return The distance field smoothing factor for this font. */
+	public float getDistanceFieldSmoothing () {
+		return distanceFieldSmoothing;
+	}
+
+	/**
+	 * @param distanceFieldSmoothing Set the distance field smoothing factor for this font. */
+	public void setDistanceFieldSmoothing (float distanceFieldSmoothing) {
+		this.distanceFieldSmoothing = distanceFieldSmoothing;
+	}
+	
+	/**
+	 * @return ShaderProgram Returns a singleton instance for the distance field shader, SpriteBatch needs to have this shader set for rendering distance field fonts. */
+	public static ShaderProgram getDistanceFieldShader() {
+		if (shader == null) {
+			shader = createDistanceFieldShader();
+			shader.setUniformf(SMOOTHING_UNIFORM, 0);
+		}
+		
+		return shader;
+	}
+	
+	/** Returns a new instance of the distance field shader, see https://github.com/libgdx/libgdx/wiki/Distance-field-fonts if the u_smoothing uniform > 0.0.
+	 *  Otherwise the same code as the default SpriteBatch shader is used.*/
+	private static ShaderProgram createDistanceFieldShader () {
+		String vertexShader = "attribute vec4 " + ShaderProgram.POSITION_ATTRIBUTE + ";\n" //
+			+ "attribute vec4 " + ShaderProgram.COLOR_ATTRIBUTE + ";\n" //
+			+ "attribute vec2 " + ShaderProgram.TEXCOORD_ATTRIBUTE + "0;\n" //
+			+ "uniform mat4 u_projTrans;\n" //
+			+ "varying vec4 v_color;\n" //
+			+ "varying vec2 v_texCoords;\n" //
+			+ "\n" //
+			+ "void main()\n" //
+			+ "{\n" //
+			+ "   v_color = " + ShaderProgram.COLOR_ATTRIBUTE + ";\n" //
+			+ "   v_color.a = v_color.a * (255.0/254.0);\n" //
+			+ "   v_texCoords = " + ShaderProgram.TEXCOORD_ATTRIBUTE + "0;\n" //
+			+ "   gl_Position =  u_projTrans * " + ShaderProgram.POSITION_ATTRIBUTE + ";\n" //
+			+ "}\n";
+		
+		String fragmentShader = "#ifdef GL_ES\n"
+			+ "	precision mediump float;\n"
+			+ "	precision mediump int;\n"
+			+ "#endif\n"
+			+ "\n"
+			+ "uniform sampler2D u_texture;\n"
+			+ "uniform float u_smoothing;\n"
+			+ "varying vec4 v_color;\n" //
+			+ "varying vec2 v_texCoords;\n" //
+			+ "\n"
+			+ "void main() {\n"
+			+ " 		if (u_smoothing > 0.0) {\n"
+			+ "			float smoothing = 0.25 / u_smoothing;\n"
+			+ "			float distance = texture2D(u_texture, v_texCoords).a;\n"
+			+ "			float alpha = smoothstep(0.5 - smoothing, 0.5 + smoothing, distance);\n"
+			+ "			gl_FragColor = vec4(v_color.rgb, alpha * v_color.a);\n"
+			+ "		}\n"
+			+ "		else {\n"
+			+ "			gl_FragColor = v_color * texture2D(u_texture, v_texCoords);\n"
+			+ "		}\n"
+			+ "}\n";
```

Comment:
Have you run the libgdx source formatter in Eclipse? Without `//` at the end of lines of `//@off`, the formatter will wreck how this code looks.